gui/system/models.py

```diff
@@ -772,11 +772,31 @@ class CertificateBase(Model):
         "CertificateAuthority",
         blank=True,
         null=True,
-        verbose_name=_("Signing Certificate Authority")
+        verbose_name=_("Signing Certificate Authority"),
+        on_delete=models.CASCADE
     )
     cert_chain = models.BooleanField(
         default=False,
     )
+    cert_acme = models.ForeignKey(
+        'ACMERegistration',
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True
+    )
+    cert_expire = models.DateTimeField(
+        blank=True,
+        null=True,
+
+    )
+    cert_acme_uri = models.URLField(
+        null=True,
+        blank=True
+    )
+    cert_domain_authenticators = EncryptedDictField(
+        null=True,
+        blank=True
+    )
```

Comment:
Why this field is used? Is it a dns.authenticator entry? If so, why this is not just a foreign key?

Reply:
So if there is a cert which supports two domains with two differnt dns providers. Say first is abc and second is xyz. So this dict would be {'abc': 1, 'xyz': 2}. It is too keep track of which authenticator is to be used for which domain

Reply:
I see, so perhaps when a dns.authenticator is deleted we should look for existing references of them here and wipe them?

Reply:
Also `cert_domain_authenticators` should probably be renamed to cert_domains_authenticators ? since there can be many?

Reply:
So before actually going ahead with authenticators I check if they exist or not when issuing a certificate. verrors are raised which tell the user he/she should set a valid dns authenticator as the provided id for this domain is not valid. I don't think we should delete the cert when a dns authenticator is deleted